### PR TITLE
Issue#600: Introduce sentry for error tracking

### LINKED ISSUE #600: Sentry
Add Sentry to the project

- [x] Add simple SentryModule
- [x] Configure with env & user details
- [ ] Figure out where to store it: maybe in an `analytics` library
- [x] Use in every app + add app info in the event info
- [x] add sentry to firebase functions

### PR DESCRIPTION
**Pull request by lsenta**
- [x] Add simple SentryModule
- [x] Configure with env & user details
- [x] Figure out where to store it: maybe in an `analytics` library -> stored in utils to make it part of every app easily.
- [x] Use in every app + add app info in the event info
- [x] add sentry to firebase functions

Fixes #600

### CODE REVIEW

File: libs/utils/src/lib/analytics.module.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  handleError(error) {
+    const eventId = Sentry.captureException(error.originalError || error);
+    const { uid, email } = this.authQuery.user || { uid: undefined, email: undefined };
+    Sentry.showReportDialog({ eventId, user: { email } });
+  }
+}
+
+// Init and add the Sentry ErrorHandler if and only if sentry is defined in env.
+if (sentryDsn) {
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/analytics.module.ts]: Better not import the module at all than verify than push a method with no provider.
In `app.module`, do : 
```typescript
import: [ sentryDsn ? AnalyticsModule : [] ]
```

File: libs/utils/src/lib/analytics.module.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import { ErrorHandler, Injectable, NgModule } from '@angular/core';
+import { sentryDsn } from '@env';
+import * as Sentry from '@sentry/browser';
+import { AuthQuery } from '@blockframes/auth/+state/auth.query';
+
+// Analytics
+const providers: any[] = [];
+
+@Injectable()
+export class SentryErrorHandler implements ErrorHandler {
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/analytics.module.ts]: Don't listen to every changes of the user 
```typescript
export class SentryErrorHandler implements ErrorHandler  {
  constructor(authQuery: AuthQuery) {
    authQuery.user$.pipe(
      filter(user => !!user), // If you don't want to catch logout
      // distinctUntilChanged((prev, curr) => /* Check if email, uid, name or username changed */)
    ).subscribe(user => {
      
    })
  }
}
```

File: libs/utils/src/lib/analytics.module.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  Sentry.init({
+    dsn: sentryDsn
+  });
+  providers.push({ provide: ErrorHandler, useClass: SentryErrorHandler });
+}
+
+@NgModule({
+  providers
+})
+export class AnalyticsModule {}
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/analytics.module.ts]: Call it `SentryErrorModule` or something more specific than `AnalyticsModule`.
Expect if we want to have one module to manage all `Analytics` tools, but then let's do that in another PR and use this one only for sentry in-between.

File: libs/utils/src/lib/sentry.module.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+  handleError(error) {
+    const eventId = Sentry.captureException(error.originalError || error);
+    const { uid, email } = this.authQuery.user || { uid: undefined, email: undefined };
+    Sentry.showReportDialog({ eventId, user: { email } });
+  }
+}
+
+// Init and add the Sentry ErrorHandler if and only if sentry is defined in env.
+if (sentryDsn) {
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/sentry.module.ts]: Is it not a duplicate of doing : `sentryDsn ? SentryModule : []` in the app modules ?

Comment (Mussuranae) [reply libs/utils/src/lib/sentry.module.ts]: It is, I removed it
